nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/ConsumeAzureEventHub.java

```diff
@@ -381,313 +366,285 @@ protected Collection<ValidationResult> customValidate(ValidationContext validati
-            return eventProcessor;
+            isRecordWriterSet = StringUtils.isNotEmpty(newValue);
         }
     }
 
-    public class EventProcessor implements IEventProcessor {
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSessionFactory sessionFactory) {
+        if (eventProcessorClient == null) {
+            eventProcessorClient = createClient(context);
+            eventProcessorClient.start();
 
-        @Override
-        public void onOpen(PartitionContext context) throws Exception {
-            getLogger().info("Consumer group {} opened partition {} of {}",
-                    new Object[]{context.getConsumerGroupName(), context.getPartitionId(), context.getEventHubPath()});
-        }
+            processSessionFactory = sessionFactory;
 
-        @Override
-        public void onClose(PartitionContext context, CloseReason reason) throws Exception {
-            getLogger().info("Consumer group {} closed partition {} of {}. reason={}",
-                    new Object[]{context.getConsumerGroupName(), context.getPartitionId(), context.getEventHubPath(), reason});
+            readerFactory = context.getProperty(RECORD_READER).asControllerService(RecordReaderFactory.class);
+            writerFactory = context.getProperty(RECORD_WRITER).asControllerService(RecordSetWriterFactory.class);
         }
 
-        @Override
-        public void onEvents(PartitionContext context, Iterable<EventData> messages) throws Exception {
-            final ProcessSession session = processSessionFactory.createSession();
-
-            try {
-
-                final StopWatch stopWatch = new StopWatch(true);
-
-                if (readerFactory != null && writerFactory != null) {
-                    writeRecords(context, messages, session, stopWatch);
-                } else {
-                    writeFlowFiles(context, messages, session, stopWatch);
-                }
+        // After a EventProcessor is registered successfully, nothing has to be done at onTrigger
+        // because new sessions are created when new messages are arrived by the EventProcessor.
+        context.yield();
+    }
 
-                // Commit NiFi first.
-                // If creating an Event Hub checkpoint failed, then the same message can be retrieved again.
-                session.commitAsync(context::checkpoint);
-            } catch (Exception e) {
-                getLogger().error("Unable to fully process received message due to " + e, e);
-                // FlowFiles those are already committed will not get rollback.
-                session.rollback();
-            }
+    @OnStopped
+    public void stopClient() {
+        if (eventProcessorClient != null) {
+            eventProcessorClient.stop();
+            eventProcessorClient = null;
+            processSessionFactory = null;
+            readerFactory = null;
+            writerFactory = null;
```

Comment:
I recommend calling the `close()` method in a try block and nulling the fields even if the `close() `fails. Otherwise `onTrigger()` can never reinitalize the client because it is not null.